Source/CarthageKit/Xcode.swift

```diff
@@ -268,6 +268,62 @@ public func schemesInProject(project: ProjectLocator) -> SignalProducer<String,
 		.map { (line: String) -> String in line.stringByTrimmingCharactersInSet(NSCharacterSet.whitespaceCharacterSet()) }
 }
 
+/// Finds schemes of projects or workspace, which Carthage should build, found
+/// within the given directory.
+public func buildableSchemesByProjectLocatorInDirectory(directoryURL: NSURL, withConfiguration configuration: String, forPlatforms platforms: Set<Platform> = []) -> SignalProducer<(ProjectLocator, [String]), CarthageError> {
+	precondition(directoryURL.fileURL)
+
+	return locateProjectsInDirectory(directoryURL)
+		.flatMap(.Concat) { project -> SignalProducer<(ProjectLocator, [String]), CarthageError> in
+			return schemesInProject(project)
+				.flatMap(.Merge) { scheme -> SignalProducer<String, CarthageError> in
+					let buildArguments = BuildArguments(project: project, scheme: scheme, configuration: configuration)
+
+					return shouldBuildScheme(buildArguments, platforms)
+						.filter { $0 }
+						.map { _ in scheme }
+				}
+				.collect()
+				.flatMapError { error in
+					if case .NoSharedSchemes = error {
+						return .init(value: [])
+					} else {
+						return .init(error: error)
+					}
+				}
+				.map { (project, $0) }
+		}
+}
+
+/// Sends pairs of a scheme and a project, the scheme actually resides in
+/// the project.
+public func schemesToBuildOfProjectLocators(projects: [(ProjectLocator, [String])]) -> SignalProducer<[(String, ProjectLocator)], CarthageError> {
```

Comment:
And call this `schemesInProjects`? This doesn't really specify anything about what's build able.